### PR TITLE
feat: Change Network

### LINKED ISSUE #77: feature request: Change Network
The example app should have an option to Change Network by calling `SetRemote` and `SetChainID`. This is similar to the option in Adena shown below. After adding this option, we should change the default remote to 127.0.0.1.
https://github.com/gnolang/gnomobile/blob/26b122530f7dc7669d56388fac1e4e846e9f472f/service/config.go#L76

Screenshot 2023-11-17 at 09 10 25

### PR DESCRIPTION
**Pull request by iuricmp**
Adds a new capability of selecting and switching networks.

fix #77

### CODE REVIEW

**Comment by iuricmp** [comment]
We need to find a way to get/detect the current selected network.

**Comment by jefft0** [comment]
> We need to find a way to get/detect the current selected network.

Thanks to another quick review by D4ryl00, PR https://github.com/gnolang/gnomobile/pull/79 is merged to main. You can rebase and use getRemote and getChainID.

File: gnoboard/src/resources/chains/chains.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+[
+  {
+    "chainId": "test3",
+    "chainName": "Testnet 3",
+    "gnoUrl": "test3.gno.land"
```
Comment (jefft0) [inline gnoboard/src/resources/chains/chains.json:5]: Should be "test3.gno.land:36657"

File: gnoboard/src/resources/chains/chains.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  {
+    "chainId": "local",
+    "chainName": "Localhost",
+    "gnoUrl": "127.0.0.1:26657"
+  },
+  {
+    "chainId": "teritori-1",
+    "chainName": "Teritori-1",
+    "gnoUrl": "testnet.gno.teritori.com"
```
Comment (jefft0) [inline gnoboard/src/resources/chains/chains.json:20]: Should be "testnet.gno.teritori.com:26657"

File: gnoboard/src/resources/chains/chains.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  {
+    "chainId": "dev.gnoswap",
+    "chainName": "Gnoswap-Testnet",
+    "gnoUrl": "dev.rpc.gnoswap.io"
```
Comment (jefft0) [inline gnoboard/src/resources/chains/chains.json:25]: I can't ping dev.rpc.gnoswap.io . Is it up? What is the port number?

File: gnoboard/src/native_modules/types.ts [hunk 4]
```diff
@@ -10,4 +10,10 @@ export type GnoConfig = {
   Mnemonic: string;
 };
 
+export type NetworkMetainfo = {
+  chainId: string;
+  chainName: string;
+  gnoUrl: string;
```
Comment (jefft0) [inline gnoboard/src/native_modules/types.ts:16]: For me, "gnoUrl" is a confusing name. A URL has something like "http://", but the Gno remote doesn't have this. Maybe "gnoAddress"?

**Review by jefft0 — APPROVED** [review]
Tested on Android and iOS simulators. It would be good to have an option to set up a custom network, but this is just an example app and we can add that later. Other projects are higher priority.

**Pull request merged**